lib/hammer_cli/defaults.rb

```diff
@@ -100,9 +100,13 @@ def switch_to_name(opt)
   end
 
   def self.defaults
-    @defaults ||= Defaults.new(HammerCLI::Settings.settings[:defaults])
-
+    return @defaults if @defaults
+    settings = HammerCLI::Settings
+    default_values = settings.get(:use_defaults) ? settings.get(:defaults) : {}
+    @defaults = Defaults.new(default_values)
   end
 
-  HammerCLI::MainCommand.subcommand "defaults", _("Defaults management"), HammerCLI::DefaultsCommand
+  if HammerCLI::Settings.get(:use_defaults)
```

Comment:
I think we should keep the command in. I find it confusing that a command would disappear when you pass `--no-defaults`

Reply:
@tstrachota Removing commands was easy way to prevent the `defaults.yaml` from rewriting when user stores new value with defaults disabled. I can make it fail on attempt to store the defaults. Would that be better?

Reply:
I'd vote for fail on attempt and showing user a some kind of warning?

Reply:
IMHO it's totally fine to save and list defaults even `--no-use-defaults` is passed. The option should just deactivate using the defaults in requests.